### ISSUE TITLE
MongoDB download not working for new versions

### ISSUE THREAD

**Issue by nileshmali** [nodkz/mongodb-memory-server#55]
When I try MongoDB version 3.6.3 on Win 7 64 bit with or without SSL, It did not work.
Seems like dowload URL scheme has changed on server.

**Comment by rcalabro** [comment]
I am getting a 403 .... checked in browser and the url has changed

```
Mongo[54984] Starting MongoDB instance with following options: {"port":54984,"dbName":"test","uri":"mongodb://localhost:54984/test","storageEngine":"ephemeralForTest","dbPath":"/var/folders/ly/5ld4fdyn33d0zb00__75n_q40000gn/T/mongo-mem-43094W4I42INCceiq"} +0ms
Downloading MongoDB: https://downloads.mongodb.org/osx/mongodb-osx-x86_64-3.6.3.tgz
  Mongo[54984] renamed /Users/***/.mongodb-binaries/mongodb-osx-x86_64-3.6.3.tgz.downloading to /Users/***/.mongodb-binaries/mongodb-osx-x86_64-3.6.3.tgz +265ms
(node:43094) UnhandledPromiseRejectionWarning: StatusCodeError: 403 - "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>B7DE450EEECE44FC</RequestId><HostId>CDNqYJFE+Fklv6AXKoaoCDqp2Z63S6d5vzVZx0Da7nmg/GHakCw1O6Pzkn/FeUeJ0JTivGLePsk=</HostId></Error>"
    at new StatusCodeError (/Users/***i/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/Users/***/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/Users/***/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/Users/***/node_modules/request/request.js:186:22)
    at Request.emit (events.js:180:13)
    at Request.<anonymous> (/Users/***/node_modules/request/request.js:1163:10)
    at Request.emit (events.js:180:13)
    at IncomingMessage.<anonymous> (/Users/***/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:272:13)
    at IncomingMessage.emit (events.js:185:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:114:19)
(node:43094) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:43094) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```


New URL: https://fastdl.mongodb.org/osx/mongodb-osx-ssl-x86_64-3.6.3.tgz

**Comment by nodkz** [comment]
Thanks for reporting.
Please try new 1.7.1 version from npm.

**Comment by nileshmali** [comment]
@nodkz This is still not working for windows build. Version 3.6.3 of windows build is only published with ssl, but URL formed by mongodb-memory-server does not match with actual download URL.
**Correct URL**
`http://fastdl.mongodb.org/win32/mongodb-win32-x86_64-2008plus-ssl-3.6.3.zip`
**Generated URL**
`https://fastdl.mongodb.org/win32/mongodb-win32-x86_64-ssl-3.6.3.zip` 

And hence it throws 403: Access Denied

**Comment by nodkz** [comment]
@nileshmali  please test 1.7.2 version

**Comment by rcalabro** [comment]
Still not working 

```
Autostarting MongoDB instance...
  Mongo[55391] Starting MongoDB instance with following options: {"port":55391,"dbName":"test","uri":"mongodb://localhost:55391/test","storageEngine":"ephemeralForTest","dbPath":"/var/folders/ly/5ld4fdyn33d0zb00__75n_q40000gn/T/mongo-mem-60977kzEyRSv1n7pN"} +0ms
Downloading MongoDB: https://fastdl.mongodb.org/osx/mongodb-osx-x86_64-3.6.3.tgz
  Mongo[55391] renamed /Users/******/.mongodb-binaries/mongodb-osx-x86_64-3.6.3.tgz.downloading to /Users/******/.mongodb-binaries/mongodb-osx-x86_64-3.6.3.tgz +162ms
StatusCodeError: 403 - "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>764806723E8FE66F</RequestId><HostId>hudw4uLqgryH0P+WT80VK92xSNoyP5nilO0AGp1a3A3G4BigbIWRRu0rvv4ybaoojAqzJhaC4KA=</HostId></Error>"
    at new StatusCodeError (/Users/******/Projects/***/***-api/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/Users/******/Projects/***/***-api/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/Users/******/Projects/***/***-api/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/Users/******/Projects/***/***-api/node_modules/request/request.js:186:22)
    at Request.emit (events.js:180:13)
    at Request.<anonymous> (/Users/******/Projects/***/***-api/node_modules/request/request.js:1163:10)
    at Request.emit (events.js:180:13)
    at IncomingMessage.<anonymous> (/Users/******/Projects/***/***-api/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:272:13)
    at IncomingMessage.emit (events.js:185:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:114:19)
npm ERR! Test failed.  See above for more details.
```

Why is this issue closed?

**Comment by nodkz** [comment]
Reopen it. You may see last commits and make PR with proper fixes. I can accept them and publish new version from mobile.
Out of work right now.

PS. Issue was closed automatically via commit message.

**Comment by nodkz** [comment]
@rcalabro changes should be made in this file
https://github.com/nodkz/mongodb-memory-server/blob/5dc0b3d022ca41d15bfdea481461489b79fd1e07/src/util/MongoBinaryDownloadUrl.js#L61

It should be very easy. Don't forget to change test file https://github.com/nodkz/mongodb-memory-server/blob/5dc0b3d022ca41d15bfdea481461489b79fd1e07/src/util/__tests__/MongoBinaryDownloadUrl-test.js#L7

**Comment by rcalabro** [comment]
Works with ssl: true on binary config

**Comment by rcalabro** [comment]
I can submit this PR probably on Friday

**Comment by nodkz** [comment]
@rcalabro don't worry we do this PR themselves.

@YozhikM can you make PR for determining proper version of Osx version with tests.
What need to do:
- completely remove ssl configuration option from code and readme
- https://github.com/nodkz/mongodb-memory-server/blob/5dc0b3d022ca41d15bfdea481461489b79fd1e07/src/util/MongoBinaryDownloadUrl.js#L63 `if (this.ssl)` change on `if(!(this.version.indexOf('3.0')===0 || this.version.indexOf('2.')===0 || this.version.indexOf('1.')===0))`

In other words for Osx distros for version above 3.1.0 we use ssl links, but below including 3.0.x without ssl. Mongodb distro has some mess with ssl build in v3.

For information, all avaliable links of mongodb binaries:
Osx https://www.mongodb.org/dl/osx
Linux https://www.mongodb.org/dl/linux
Win https://www.mongodb.org/dl/win32
Will be great if you add these links as comment in the code for quick solving future problems.

Thanks.

**Comment by nileshmali** [comment]
Working now on windows. Hence closing the issue.

**Comment by austi10** [comment]
> **Correct URL**
> `http://fastdl.mongodb.org/win32/mongodb-win32-x86_64-2008plus-ssl-3.6.3.zip`
> **Generated URL**
> `https://fastdl.mongodb.org/win32/mongodb-win32-x86_64-ssl-3.6.3.zip`
> 
> And hence it throws 403: Access Denied
This issue still persists for me upon npm install

**Comment by hasezoey** [comment]
@austi10 thanks for reporting, because this issue (#55) is already old and closed, we will use https://github.com/nodkz/mongodb-memory-server/issues/399 instead to track this
